### PR TITLE
Upgrade Bridge and Pulumi versions. Integrate dependencies from master branch

### PR DESCRIPTION
**Pull request by guineveresaenger**
This PR converges the new `main` branch with the old `master` branch.

In preparation for this PR, the `v5` branch(with PF changes) was re-named to `main`.

Then all commits from `master` (with dependency bumps, modern pulumi and bridge versions) were reconciled with `main` on this upgrade branch via merge:
- update pulumi version
- update bridge version
- modernize PF code in resources.go
- keep all dependency bumps and new pulumi features such as ESC
- re-generate schema
- re-generate SDKs

After this pull request merges, we will:
- make `main` the default branch for this repository
- rename the old `master` branch to `v4`
- migrate this repository to a patched repository and upgrade the upstream version as per #341., which represents a major upgrade v4 --> v5. (Note that the code in this PR already makes the v5 changes.)

### CODE REVIEW

File: .ci-mgmt.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+provider: tls
+major-version: 5
+makeTemplate: bridged
+# TODO: Enable this once our tls provider is updated with framework support
+# generate-nightly-test-workflow: true
```
Comment (iwahbe) [inline .ci-mgmt.yaml:5]: Shouldn't this be enabled at this point. The provider has framework support.

File: provider/resources.go [hunk 2]
```diff
@@ -59,17 +60,21 @@ func tlsResource(mod string, res string) tokens.Type {
 	return tlsType(mod+"/"+fn, res)
 }
 
+var metadata []byte
```
Comment (t0yv0) [inline provider/resources.go:63]: Shouldn't this have embedded data?

**Review by t0yv0 — APPROVED** [review]

**Pull request merged**